src/factory/factory.cpp

```diff
@@ -70,9 +70,6 @@ Factory::CreateIndex(const std::string& origin_name,
             logger::debug("created a diskann index");
             return std::make_shared<DiskANN>(diskann_params, index_common_params);
         } else if (name == INDEX_HGRAPH) {
-            if (allocator == nullptr) {
-                index_common_params.allocator_ = DefaultAllocator::Instance().get();
-            }
```

Comment:
why can we remove this check now?

Reply:
allocator will not be nullptr, it's initialize at common param parse

Reply:
For params checking, we need throw an invalid_argument exception when `allocator` not set ?